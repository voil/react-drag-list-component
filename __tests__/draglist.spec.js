/*
 * =============================================================================
 * Project: react-drag-list-component
 * Created Date: 2018-03-26, 13:33:51
 * Author: Przemysław Drzewicki <przemyslaw.drzewicki@gmail.com>
 * =============================================================================
 * Last Modified: 2018-03-28, 11:02:18
 * Modified By: Przemysław Drzewicki
 * =============================================================================
 * Copyright (c) 2018 webonweb
 * =============================================================================
 */
import React from 'react';
import DragList from '../src/lib';


const target = document.createElement( "div" );
const currentTarget = { dataset : { index: 1 } };
const wrapper = shallow(<DragList 
  list={[{ name: 'test' }]}
  update={(item, old, next) => {}}
  render={(item, index) => <div key={index}>{item.name}</div>}/>);

test('DragList render component correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('Test current element is set', () => {
  const elementList = wrapper.find('div').at(1);
  elementList.simulate('mousedown', { target, currentTarget })

  expect(wrapper.state().current).not.toBe(null);
})

test('Check is placeholder default is set to true', () => {
  expect(wrapper.state().placeholder).toBe(true);
})

test('Check is placeholder hidden', () => {
  const wrapperPlaceholder = shallow(<DragList 
    list={[{ name: 'test' }]}
    placeholder={false}
    update={(item, old, next) => {}}
    render={(item, index) => <div key={index}>{item.name}</div>}/>);

  expect(wrapperPlaceholder.state().placeholder).toBe(false);
})

test('Test target element is set', () => {
  const elementList = wrapper.find('div').at(1);
  elementList.simulate('mousedown', { target, currentTarget });

  elementList.simulate('mousemove', { target, currentTarget });
  expect(wrapper.state().target).not.toBe(null);
})

test('Test current after mouse up', () => {
  const elementList = wrapper.find('div').at(1);
  elementList.simulate('mousedown', { target, currentTarget });

  elementList.simulate('mouseup', { target, currentTarget });

  expect(wrapper.state().current).toBe(null);
  expect(wrapper.state().target).toBe(null);
});