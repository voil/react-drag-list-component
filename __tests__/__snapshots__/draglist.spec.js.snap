// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DragList render component correctly 1`] = `
<div
  className="drag-list"
>
  <div
    data-index={0}
    key="drag-list-0"
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
  >
    <div
      key="0"
    >
      test
    </div>
  </div>
  <div
    className="drag-list-current"
    onMouseUp={[Function]}
    style={
      Object {
        "left": 0,
        "top": 0,
      }
    }
  />
</div>
`;
